updater/fetchers/alpine/alpine.go

```diff
@@ -0,0 +1,177 @@
+	ErrFilesystem = errors.New("updater/fetchers: something went wrong when interacting with the fs")
+
+	// ErrGitFailure is returned when a fetcher fails to interact with git.
+	ErrGitFailure = errors.New("updater/fetchers: something went wrong when interacting with git")
+
+	log = capnslog.NewPackageLogger("github.com/coreos/clair", "updater/fetchers/alpine")
+)
+
+func init() {
+	updater.RegisterFetcher("alpine", &fetcher{})
+}
+
+type fetcher struct {
+	repositoryLocalPath string
+}
+
+func (f *fetcher) FetchUpdate(db database.Datastore) (resp updater.FetcherResponse, err error) {
+	log.Info("fetching Alpine vulnerabilities")
+
+	// Pull the master branch.
+	var commit string
+	commit, err = f.pullRepository()
+	if err != nil {
+		return
+	}
+
+	// Ask the database for the latest commit we successfully applied.
+	var dbCommit string
+	dbCommit, err = db.GetKeyValue("alpineUpdater")
+	if err != nil {
+		return
+	}
+
+	// Append any changed vulnerabilities to the response.
+	if commit == dbCommit {
+		log.Debug("no alpine update")
+	} else {
+		for _, namespace := range []string{"v3.3", "v3.4"} {
+			file, err := os.Open(f.repositoryLocalPath + "/" + namespace + "/main.yaml")
+			if err != nil {
+				// TODO
+			}
+
+			vulns, err := parseYAML(file)
+			if err != nil {
+				// TODO
+			}
+			resp.Vulnerabilities = append(resp.Vulnerabilities, vulns...)
+			file.Close()
+		}
+	}
+
+	resp.FlagName = updaterFlag
+	resp.FlagValue = commit
+	return
+}
+
+func (f *fetcher) pullRepository() (commit string, err error) {
+	if _, pathExists := os.Stat(f.repositoryLocalPath); f.repositoryLocalPath == "" || os.IsNotExist(pathExists) {
+		if f.repositoryLocalPath, err = ioutil.TempDir(os.TempDir(), "alpine-secdb"); err != nil {
+			return "", ErrFilesystem
+		}
+
+		if out, err := utils.Exec(f.repositoryLocalPath, "git", "pull"); err != nil {
+			f.Clean()
+			log.Errorf("could not pull alpine-secdb repository: %s. output: %s", err, out)
+			return "", cerrors.ErrCouldNotDownload
+		}
+	}
+
+	out, err := utils.Exec(f.repositoryLocalPath, "git", "rev-parse", "HEAD")
+	if err != nil {
+		return "", ErrGitFailure
+	}
+
+	commit = strings.TrimSpace(string(out))
+	return
+}
+
+func (f *fetcher) Clean() {
+	if f.repositoryLocalPath != "" {
+		os.RemoveAll(f.repositoryLocalPath)
+	}
+}
+
+type secdbFile struct {
+	Distro   string `yaml:"distroversion"`
+	Packages []struct {
+		Pkg struct {
+			Name    string   `yaml:"name"`
+			Version string   `yaml:"ver"`
+			Fixes   []string `yaml:"fixes"`
+		} `yaml:"pkg"`
+	} `yaml:"packages"`
+}
+
+func parseYAML(r io.Reader) (vulns []database.Vulnerability, err error) {
+	var rBytes []byte
+	rBytes, err = ioutil.ReadAll(r)
+	if err != nil {
+		return
+	}
+
+	var file secdbFile
+	err = yaml.Unmarshal(rBytes, &file)
+	if err != nil {
+		return
+	}
+	for _, pack := range file.Packages {
+		pkg := pack.Pkg
+		for _, fix := range pkg.Fixes {
+			version, err := types.NewVersion(pkg.Version)
+			if err != nil {
+				continue
```

Comment:
log.Warningf("could not parse package version '%s': %s. skipping", version, err.Error())